samples/react-timeline/src/webparts/timeline/components/Popup/TimelineEvent.tsx

```diff
@@ -0,0 +1,379 @@
+      let timeLineDate: Date = moment(event.acivityDate).toDate();
+
+      // Update Component Data
+      this.setState({
+        eventData: event,
+        acivityDate: timeLineDate,
+        startSelectedHour: { key: startHour, text: startHour },
+        startSelectedMin: { key: startMin, text: startMin },
+        activityDescription: event.activityDescription,
+        activityTitle: event.activityTitle,
+        activityLink: event.activityLink,
+        activityPictureUrl: event.activityPictureUrl,
+        isloading: false
+      });
+    }
+    else {
+      this.setState({
+        acivityDate: new Date(),
+        activityDescription: '',
+        activityTitle: '',
+        activityLink: '',
+        activityPictureUrl: '',
+        isloading: false,
+        eventData: { ...event },
+      });
+    }
+  }
+
+  public async componentDidMount() {
+    await this.renderEventData();
+  }
+
+  private onStartChangeHour = (ev: React.FormEvent<HTMLDivElement>, item: IDropdownOption): void => {
+    this.setState({ startSelectedHour: item });
+  }
+
+  private _onEventTitleChange = (ev: any, newText: string): void => {
+    this.setState({ eventData: { ...this.state.eventData, activityTitle: newText } });
+  }
+
+  private _onActivityPictureURLChange = (ev: any, newText: string): void => {
+    this.setState({ eventData: { ...this.state.eventData, activityPictureUrl: newText } });
+  }
+
+  private _onActivityLinkURLChange = (ev: any, newText: string): void => {
+    this.setState({ eventData: { ...this.state.eventData, activityLink: newText } });
+  }
+
+  private onStartChangeMin = (ev: React.FormEvent<HTMLDivElement>, item: IDropdownOption): void => {
+    this.setState({ startSelectedMin: item });
+  }
+
+  private onDescriptionChange = (e): void => {
+    this.setState({ activityDescription: e.target.value });
+  }
+
+  private onGetErrorMessageTitle(value: string): string {
+    let returnMessage: string = '';
+
+    if (value.length === 0) {
+      returnMessage = "Error";
+    }
+    else {
+      this.setState({ eventData: { ...this.state.eventData, activityTitle: value }, disableButton: false, errorMessage: '' });
+    }
+    return returnMessage;
+  }
+
+  private onDelete(ev: React.MouseEvent<HTMLDivElement>) {
+    ev.preventDefault();
+    this.setState({ displayDialog: true });
+  }
+
+  private closeDialog = (): void => {
+    this.setState({ displayDialog: false });
+  }
+
+  private onSelectDateStart(newDate: Date) {
+    this.setState({ acivityDate: newDate });
+  }
+
+  public render(): React.ReactElement<IEventProps> {
+    return (
+      <div>
+        <Dialog
+          isOpen={this.props.showPanel}
+          closeButtonAriaLabel="Close"
+          dialogContentProps={{
+            type: DialogType.normal,
+            title:
+              this.props.panelMode == 2
+                ? "Edit Timeline Event"
+                : "Create Timeline Event",
```

Comment:
Seems like you have an opportunity to use your localization files to localize the content in this class.